xmake/core/base/table.lua

```diff
@@ -428,9 +428,9 @@ function table.keys(tbl)
 end
 
 -- get order keys of a table
-function table.orderkeys(tbl)
+function table.orderkeys(tbl, callback)
```

Comment:
it will break some thing

https://github.com/xmake-io/xmake/blob/23f598d853e16c25360b70a082629bdd0d35eedd/xmake/core/project/target.lua#L1892

```lua
table.orderpairs(self:sourcebatches())
```

`self:sourcebatches()` will return 2 values.

Reply:
Ah, I was wondering why the rules test failed. Do you have a workaround in mind?
If not, I'll close this.

Reply:
You need to fix all code that calls orderpairs, if it's arg return 2 args.

```lua
orderpairs((self:sourcebatches()))
```